### DISCUSSION TITLE
convert written numbers to numbers

### DISCUSSION THREAD

**Discussion by Karim-Mane** [epiverse-trace/cleanepi#17] [Q&A]
I have an `age` column that contains 90% of numeric values and 7% of character values (these are numbers written in letters). I am looking for an R package that could convert the `character` values into `numeric`. 
I found the `{wordstonumbers}` package in GitHub that does what I need. Given this, I have the following options:
1. To copy the function and cite the author as a copy right holder given that we are using the same License type. This needs the author's consent.
2. To wait until the package is in CRAN if I want to use this function before submitting {cleanepi} in CRAN.
3. To create my own function/package for this purpose.
4. To work together with the author and get his package on CRAN and be able to use it in our package. 

What do you advise on this?

**Comment by adamkucharski** [comment]
How complex would the required function be? My initial feeling is (3) preference rather than taking on a dependency for something that is relatively simple to implement from scratch (intuitively this task seems to just require some vector matching and standardisation)? But @Bisaloo  may have more thoughts here on what we're doing elsewhere.

**Comment by Bisaloo** [comment]
I prefer to have a dedicated package for this, and use it as a dependency, because this seems like a task somewhat independent from cleanepi core functionality and the specific feature might be useful for others who don't want to pull the entire cleanepi package.

It's probably possible to code it ourselves as Adam is suggesting but I would recommend investigating existing alternatives first:

- [numberwang](https://github.com/coolbutuseless/numberwang)
- [nombre](https://cran.r-project.org/package=nombre)
- [english](https://cran.r-project.org/package=english)
- [spanish](https://cran.r-project.org/package=spanish)

As you can see from this list, an important question is whether you will need this functionality in multiple languages or not. Some of these packages could be extended to include other languages but due to their different infrastructure, it's likely to be easier for some than others.

**Reply by Karim-Mane** [reply]
Thank you @Bisaloo and @adamkucharski 

This has now been addressed by {numberize}, an R package created by @bahadzie
